test/integration/src/test/java/io/pravega/test/integration/MetricsTest.java

```diff
@@ -0,0 +1,293 @@
+
+            EventStreamReader<String> reader2 = clientFactory.createReader(readerName,
+                    readerGroupName2,
+                    new JavaSerializer<String>(),
+                    ReaderConfig.builder().build());
+
+            for (int q = 0; q < TOTAL_NUM_EVENTS; q++) {
+                try {
+                    String eventRead2 = reader2.readNextEvent(SECONDS.toMillis(2)).getEvent();
+                    log.info("Reading event {}", eventRead2);
+                } catch (ReinitializationRequiredException e) {
+                    log.warn("Test Exception while reading from the stream", e);
+
+                }
+            }
+
+            long countAfterCacheEvicted = MetricsProvider.METRIC_REGISTRY.getCounters().get("pravega.segmentstore.segment_read_bytes." + scope + "." + STREAM_NAME + ".0.Counter").getCount();
+
+            //Count starts from 0, rather than adding up to previously ready bytes, as cache is evicted.
+            Assert.assertEquals(bytesWritten, countAfterCacheEvicted);
+
+            Map<Double, Double> map = new HashMap<>();
+            map.put(0.0, 1.0);
+
+            //Seal segment 0, create segment 1
+            CompletableFuture<Boolean> scaleStatus = controller.scaleStream(new StreamImpl(scope, STREAM_NAME),
+                    Collections.singletonList(0),
+                    map,
+                    executorService).getFuture();
+            Assert.assertTrue(scaleStatus.get());
+
+            EventStreamWriter<String> writer2 = clientFactory.createEventWriter(STREAM_NAME,
+                    new JavaSerializer<String>(),
+                    EventWriterConfig.builder().build());
+
+            for (int i = 0; i < TOTAL_NUM_EVENTS; i++) {
+                try {
+                    log.info("Writing event {}", event);
+                    writer2.writeEvent("", event);
+                    writer2.flush();
+                } catch (Throwable e) {
+                    log.warn("Test exception writing events: {}", e);
+                    break;
+                }
+            }
+
+            Exceptions.handleInterrupted(() -> Thread.sleep(70 * 1000));
+
+            for (int j = 0; j < TOTAL_NUM_EVENTS; j++) {
+
+                try {
+                    String eventRead2 = reader1.readNextEvent(SECONDS.toMillis(2)).getEvent();
+                    log.info("Reading event {}", eventRead2);
+                } catch (ReinitializationRequiredException e) {
+                    log.warn("Test Exception while reading from the stream", e);
+                }
+
+            }
+
+            long countFromSecondSegment = MetricsProvider.METRIC_REGISTRY.getCounters().get("pravega.segmentstore.segment_read_bytes." + scope + "." + STREAM_NAME + ".1.Counter").getCount();
+
+            Assert.assertEquals(bytesWritten, countFromSecondSegment);
+
+            //Count remains same for segment 0, as it is sealed
+            Assert.assertEquals(bytesWritten, countAfterCacheEvicted);
```

Comment:
i am sorry, i dont get this.. what are you asserting here?

Reply:
fixed it.